src/backend/distributed/shared_library_init.c

```diff
@@ -2177,12 +2198,73 @@ StatisticsCollectionGucCheckHook(bool *newval, void **extra, GucSource source)
 static void
 CitusAuthHook(Port *port, int status)
 {
+	uint64 gpid = ExtractGlobalPID(port->application_name);
```

Comment:
just curiosity.  Can portt->application_name be different from `application_name`?

Because last time I checked this, this check was `IsCitusInternalBackend`?